plugins/modules/dcnm_vrf.py

```diff
@@ -706,60 +807,49 @@ def diff_for_attach_deploy(self, want_a, have_a, replace=False):
-                                        interface_match = True
-                                        if wlite["DOT1Q_ID"]:
-                                            if (
-                                                wlite["DOT1Q_ID"]
-                                                != hlite["DOT1Q_ID"]
-                                            ):
-                                                found = False
-                                                break
-
-                                        if wlite["IP_MASK"]:
-                                            if (
-                                                wlite["IP_MASK"]
-                                                != hlite["IP_MASK"]
-                                            ):
-                                                found = False
-                                                break
-
-                                        if wlite["NEIGHBOR_IP"]:
-                                            if (
-                                                wlite["NEIGHBOR_IP"]
-                                                != hlite["NEIGHBOR_IP"]
-                                            ):
-                                                found = False
-                                                break
-
-                                        if wlite["IPV6_MASK"]:
-                                            if (
-                                                wlite["IPV6_MASK"]
-                                                != hlite["IPV6_MASK"]
-                                            ):
-                                                found = False
-                                                break
-
-                                        if wlite["IPV6_NEIGHBOR"]:
-                                            if (
-                                                wlite["IPV6_NEIGHBOR"]
-                                                != hlite["IPV6_NEIGHBOR"]
-                                            ):
-                                                found = False
-                                                break
-
-                                        if wlite["PEER_VRF_NAME"]:
-                                            if (
-                                                wlite["PEER_VRF_NAME"]
-                                                != hlite["PEER_VRF_NAME"]
-                                            ):
-                                                found = False
-                                                break
-
-                                        if found:
+                                    if wlite["IF_NAME"] != hlite["IF_NAME"]:
+                                        continue
+                                    found = True
+                                    interface_match = True
+                                    if wlite["DOT1Q_ID"]:
```

Comment:
Was thinking that maybe we could refactor these property comparisons to use a helper function.  Something along the lines of this?

```python
def compare_properties(self, want, have, property_list):

    for prop in property_list:
        if want.get(prop):
            if want[prop] != have[prop]:
                return False
    
    return True
```

```python
property_list = ["DOT1Q_ID", "IP_MASK", ...]
found = self.compare_properties(wlite, hlite, property_list)
```

Reply:
Beautiful.  Yes, let me add this Mike.

Reply:
Does the following look OK?  Adding protection on access to the have dict.

```python
    def compare_properties(self, dict1, dict2, property_list):
        """
        Given two dictionaries and a list of properties:

        - Return True if all property values match.
        - Return False otherwise
        """
        for prop in property_list:
            if dict1.get(prop) != dict2.get(prop):
                    return False
        return True
```

Reply:
Looks great!